driver/src/price_finding/naive_solver.rs

```diff
@@ -101,8 +102,14 @@ impl PriceFinding for NaiveSolver {
         orders: &[Order],
         state: &AccountState,
     ) -> Result<Solution, PriceFindingError> {
+        // Fetch max_token_id to determine num_tokens
+        let max_token_id = orders
+            .iter()
+            .map(|o| cmp::max(o.buy_token, o.sell_token))
+            .max()
+            .unwrap();
```

Comment:
why not unwrap_or(0) here?

Reply:
Yes absolutely. I am actually surprised this is passing. Will fix this and add a test.